app/services/plans/sync_new_charges_with_children_service.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Plans
+  class SyncNewChargesWithChildrenService < BaseService
+    Result = BaseResult
+    def initialize(plan:)
+      @plan = plan
+      super
+    end
+
+    def call
+      # Note: this check is only needed if child charges "lost" their parents:
+      # when a parent plan was updated without sending charges_ids, so the "original" parents were deleted
+      return result.forbidden_failure!(code: "plan_has_undistinguishable_charges") if plan_has_undistinguishable_charges?
+
+      plan.charges.each do |charge|
+        sync_charge_for_children(charge)
+      end
+    end
+
+    private
+
+    attr_reader :plan
+
+    def sync_charge_for_children(charge)
+      plan.children.joins(:subscriptions).where(subscriptions: {status: %w[active pending]}).distinct.pluck(:id).each_slice(20) do |child_ids|
+        Charges::SyncChildrenBatchJob.perform_later(
+          children_plans_ids: child_ids,
+          charge:
+        )
+      end
+    end
+
+    def plan_has_undistinguishable_charges?
```

Comment:
So if we cannot match exactly children charges, since there are multiple ones with the same BM, the feature is not applicable? Is this correct?

Reply:
yes, they will work on not sending us updates without ids, but for the billing run they want to have all charges in children plans